libbeat/cmd/setup.go

```diff
@@ -27,6 +27,19 @@ import (
 	"github.com/elastic/beats/libbeat/cmd/instance"
 )
 
+const (
+	//TemplateKey used for defining template in setup cmd
+	TemplateKey = "template"
+	//DashboardKey used for registering dashboards in setup cmd
+	DashboardKey = "dashboards"
+	//MachineLearningKey used for registering ml jobs in setup cmd
+	MachineLearningKey = "machine-learning"
+	//PipelineKey used for registering pipelines in setup cmd
+	PipelineKey = "pipelines"
+	//ILMPolicyKey used for registering ilm in setup cmd
+	ILMPolicyKey = "ilm-policy"
+)
+
 func genSetupCmd(name, idxPrefix, version string, beatCreator beat.Creator) *cobra.Command {
```

Comment:
> I still prefer the original approach, minus the bool instead of boolVars

Not really sure I understand you correct. 
As pointed out in the description, the intention of this PR is to make the `setup` cmd more flexible. The APM Server needs to be able to remove flags from this command, as e.g. we don't support setting up dashboards via the server any longer. The current solution does not let you remove commands, e.g, if only `./beat setup` is run without a flag all configured flags are set to `true`. 

In the current codebase the flag keys are repeatedly used as strings, this PR defines the keys as constants to avoid using a wrong key by accident (which could happen since they are not streamlined). 
At the moment the `beat.Setup` takes every setup flag as a separate parameter, which is pretty unflexibel for future changes. This is why I originally suggested to use a `map[string]bool` instead. I do see your point of making this even more flexible by passing in a struct. Using a struct over a map leads to some repetition (since I didn't bother to use reflection for a logic, that generally feels pretty tight to some defined keys). 

> setup.Flags().Bool(&template, false, "Setup index template")

I assume you meant to use `setup.Flags().BoolVar(&template, templateKey, false, "")` here.  

Looking at your solution, I don't see how this would work to filter out flags that are _NOT_ registered, see https://github.com/elastic/beats/pull/10277/files/115c247a852d7de35075ea8676d42904ab3b2af2#diff-bea127b87274a809c128801352efaaf3R76, which again, is the motivation of this PR.